### DISCUSSION TITLE
Ghost higher baud rate - Ghost menu

### DISCUSSION THREAD

**Discussion by benedekvajas** [EdgeTX/edgetx#1941] [Q&A] [answered]
Hi all!

I just got the ghost module, and I'm having trouble finding the optimal settings so that everything works. 

Radio: QX7 with mod installed when I got the CRSF module + 1bit mode as I've seen both recommended together.
Mode Race500

The ghost menu only works on 115k baud rate.
If I select 400k the Ghost menu hangs: "Waiting for module". Also at this point the module gets stuck in that mode and only restart helps.

Also I noticed that in 2.7 the option to select higher baud rate on the Setup page is not enabled for GHST. Only 115k and 400k.
While on 2.5 I can set the baud rate to any value on the hardvare page. I guess this is for a reason that it was moved to model page and the options vary between the protocols.

Questions:
- Am I missing out if I don't run the highest possible baud rate? I know it's 1/2ms advantage at best but still...
- Is the issue with the ghost menu script or there is some other issue on my side?

Basically does Ghost menu work for anyone else on anything other than 115k baud rate?

**Comment by pfeerick** [comment, marked as answer]
I'm only new to GHOST myself, but to the best of my knowledge, you must set it to 115k and GHST, and the onebit mode does not apply. According to JC:
> the channels will be sent at 400k, and telemetry at 115k" 

and

> the hardware mod "is useless and irrelevant for ghost module since we have introduced asymmetrical speed for ghost (the speed limit is only for data going in, not out)"

per https://www.rcgroups.com/forums/showthread.php?3692187-IRC-Ghost-Owner-s-Thread/page3

> Also I noticed that in 2.7 the option to select higher baud rate on the Setup page is not enabled for GHST. Only 115k and 400k.
While on 2.5 I can set the baud rate to any value on the hardvare page. I guess this is for a reason that it was moved to model page and the options vary between the protocols.

Correct. Different protocols want different baud rates. i.e. you shouldn't be able to alter the rate for the MPM, CRSF can vary depending on if you are using Crossfire or ELRS on the other end, and GHST really wants 115k, but *can* support 400k, but will just result in the link being idle for more than half of the time.

**Reply by benedekvajas** [reply]
Wow, thanks for the quick and comprehensive answer. I somehow missed that rcgroups discussion. Also it seems all the other relevant discussion takes place in the facebook group, that's why it was so difficult to find any help.

Anyways, thanks a lot!
